### PR TITLE
Unify various layer extension handling, big UO tidy-ups.

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
I see, travis is angry about the merge, not the state on this branch. Will rebase tomorrow.

**Comment by mark-lunarg** [comment]
Thanks Chris, this has been on the list for awhile.  As is all the intercept_xxx_command crud.  LGTM.

File: scripts/unique_objects_generator.py [hunk 1]
```diff
@@ -873,7 +873,10 @@ def WrapCommands(self):
             dispatchable_type = cmdinfo.elem.find('param/type').text
             dispatchable_name = cmdinfo.elem.find('param/name').text
             # Generate local instance/pdev/device data lookup
-            self.appendSection('command', '    layer_data *dev_data = GetLayerDataPtr(get_dispatch_key('+dispatchable_name+'), layer_data_map);')
+            if dispatchable_type in ["VkPhysicalDevice", "VkInstance"]:
+                self.appendSection('command', '    instance_layer_data *dev_data = GetLayerDataPtr(get_dispatch_key('+dispatchable_name+'), instance_layer_data_map);')
```
Comment (chrisforbes) [inline scripts/unique_objects_generator.py]: This has annoying ripples throughout the rest of the UO codegen

**Comment by chrisforbes** [comment]
Rebased to catch @krOoze's change which required a slight tweak. TravisCI should be happy now.

**Pull request merged**